File: Algorithms Advanced/others/GraphAlgorithms/Python/Floyd-Warshall.py
def floyd_warshall(graph, nodes):
    dist = [[graph[i][j] for j in range(nodes)] for i in range(nodes)]

    for k in range(nodes):
        for i in range(nodes):
            for j in range(nodes):
                if dist[i][k] != float('inf') and dist[k][j] != float('inf'):
                    dist[i][j] = min(dist[i][j], dist[i][k] + dist[k][j])

    for row in dist:
        print(" ".join(["INF" if x == float('inf') else str(x) for x in row]))

# Example
nodes = 4
graph = [
    [0, 3, float('inf'), 5],
    [2, 0, float('inf'), 4],
    [float('inf'), 1, 0, float('inf')],
    [float('inf'), float('inf'), 2, 0]
]
floyd_warshall(graph, nodes)

File: Algorithms Advanced/others/GraphAlgorithms/C++/Floyd-Warshall.cpp
#include <iostream>
#include <vector>
#include <climits>
using namespace std;

void floydWarshall(vector<vector<int>> &graph, int nodes) 
{
    vector<vector<int>> dist = graph;

    for (int k = 0; k < nodes; ++k) 
    {
        for (int i = 0; i < nodes; ++i) 
        {
            for (int j = 0; j < nodes; ++j) 
            {
                if (dist[i][k] != INT_MAX && dist[k][j] != INT_MAX && dist[i][k] + dist[k][j] < dist[i][j]) 
                {
                    dist[i][j] = dist[i][k] + dist[k][j];
                }
            }
        }
    }

    for (int i = 0; i < nodes; ++i) 
    {
        for (int j = 0; j < nodes; ++j) 
        {
            if (dist[i][j] == INT_MAX) cout << "INF ";
            else cout << dist[i][j] << " ";
        }
        cout << endl;
    }
}

int main() 
{
    int nodes = 4;
    vector<vector<int>> graph = {
        {0, 3, INT_MAX, 5},
        {2, 0, INT_MAX, 4},
        {INT_MAX, 1, 0, INT_MAX},
        {INT_MAX, INT_MAX, 2, 0}};

    floydWarshall(graph, nodes);
    return 0;
}
